ait/core/server/plugin.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+import importlib
+import re
+
+import gevent
+import gevent.monkey; gevent.monkey.patch_all()
+
+import ait.core
+from ait.core import  db, limits, log, notify, tlm
+from client import ZMQInputClient
+
+
+class Plugin(ZMQInputClient):
+    """
+    This is the parent class for all plugins. All plugins must implement
+    their own process method which is called when a message is received.
+    """
+
+    def __init__(self, inputs, outputs, zmq_args={}, **kwargs):
+        self.type = 'Plugin'
+        self.name = type(self).__name__
+        self.inputs = inputs
+        self.outputs = outputs
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        super(Plugin, self).__init__(**zmq_args)
+
+    def __repr__(self):
+        return '<Plugin name={}>'.format(self.name)
+
+    def process(self, input_data, topic=None):
+        raise NotImplementedError((
+            'This process method must be implemented by a custom plugin class '
+            'that inherits from this abstract plugin. This abstract Plugin '
+            'class should not be instantiated. This process method will be '
+            'called whenever a message is received by the plugin.'))
+
+
+class DataArchive(Plugin):
+
+    def __init__(self, inputs, outputs, datastore='ait.core.db.InfluxDBBackend', **kwargs):
+        super(DataArchive, self).__init__(inputs, outputs, **kwargs)
+
+        self.datastore = datastore
+
+    def process(self, input_data, topic=None, **kwargs):
```

Comment:
A log of this should get pulled out of process and moved to the `__init__`. Parsing the telemetry dictionary into a helpful format, setting up a database connection, etc. shouldn't be happening every time we get a packet and process it.